tests/test_regularization.py

```diff
@@ -116,3 +116,48 @@ def test_regularization_events():
     # print(to_preserve)
     assert np.mean(to_reduce) < 0.1
     assert np.mean(to_preserve) > 0.5
+
+
+def test_regularization_lagged_regressor():
+    """
+    Test case for  regularization feature of lagged regressors. Utlizes a
+    synthetic dataset with 4 noise-based lagged regressors (a, b, c, d).
+    The first and last lagged regressors (a, d) are expected to have a weight
+    close to 1. The middle lagged regressors (b, c) meanwhile are expected to
+    have a weight close to 0, due to the regularization. All other model
+    components are turned off to avoid side effects.
+    """
+    df, lagged_regressors = generate_lagged_regressor_dataset(periods=100)
+    df = df_utils.check_dataframe(df, check_y=False)
+
+    m = NeuralProphet(
+        epochs=20,
+        batch_size=8,
+        learning_rate=0.1,
+        yearly_seasonality=False,
+        weekly_seasonality=False,
+        daily_seasonality=False,
+        growth="off",
+        normalize="off",
+    )
+    m = m.add_lagged_regressor(
+        n_lags=3,
+        names=[lagged_regressor for lagged_regressor, _ in lagged_regressors],
+        regularization=0.1,
+    )
+    m.fit(df, freq="D")
+
+    lagged_regressors_config = dict(lagged_regressors)
+
+    for name in m.config_covar.keys():
+        weights = m.model.get_covar_weights(name).detach().numpy()
+        weight_average = np.average(weights)
+
+        lagged_regressor_weight = lagged_regressors_config[name]
+
+        if lagged_regressor_weight > 0.9:
+            assert weight_average > 0.6
+        else:
+            assert weight_average < 0.3
```

Comment:
shouldn't this be 0.1 without regularization? Thus it should be less than 0.1 with regularization

Reply:
We would need more `epochs` to ensure more stable values to test for `<0.1`, e.g. with `epochs=30` or better `epochs=40` instead of `epochs=20` we could certainly test for `<0.1`.

Reply:
for regularization, try similar as AR
`reg = utils_torch.penalize_nonzero(weights, eagerness=1.0, acceptance=1.0)`